packages/uui-color-area/lib/uui-color-area.element.ts

```diff
@@ -97,12 +97,17 @@ export class UUIColorAreaElement extends LitElement {
       const parsed = colord(newVal);
 
       if (parsed.isValid()) {
-        const { h, s, l } = parsed.toHsl();
+        const { h, s, l, a } = parsed.toHsl();
 
-        this.hue = h;
-        this.saturation = s;
+        // Update hue from parsed color, but when color is black, value from hue slider may be different from zero.
+        if (h !== 0) {
+          this.hue = h;
+        }
+
+        //this.saturation = s;
```

Comment:
I tried the following here, but it seem it made the handle "jump" and bit anyway and the color spectrum seemed to be affected. When dragging the handle in the spectrum, I would expect the gradient shouldn't change.

```
if (s !== 0) {
    this.saturation= s;
}
```